djoser/serializers.py

```diff
@@ -61,12 +62,9 @@ def create(self, validated_data):
         return user
 
     def perform_create(self, validated_data):
-        with transaction.atomic():
-            user = User.objects.create_user(**validated_data)
-            if settings.get('SEND_ACTIVATION_EMAIL'):
-                user.is_active = False
-                user.save(update_fields=['is_active'])
-        return user
+        if config.SEND_ACTIVATION_EMAIL:
+            validated_data.update({'is_active': False})
+        return User.objects.create_user(**validated_data)
```

Comment:
AFAIK we cannot get rid of `with transaction.atomic():` context manager here because `perform_create` method is called by `create` method where `IntegrityError` is handled. If we do not use separate transaction here then (depending on global django settings) the current transaction (e.g. per request) can be not valid anymore (e.g. for postgres).